CODEOWNERS

```diff
@@ -9,5 +9,15 @@
 # Order in this file is important. Only the last match will be
 # used. See https://help.github.com/articles/about-code-owners/
 
+# All markdown documentation needs an extra review
 *.md    @kata-containers/documentation
 
+# Kernel changes are delicate so require additional reviews.
+/kernel/ @kata-containers/kernel
```

Comment:
I *think* we are going to be OK with these - but, this is iiuc shifting the pattern matches from being pretty repo specific to being global across all repos. afaik, the 'there are no other proto files' rule and the 'anything in /kernel is directly kernel related' rules might hold - but, beware, they might start matching some unintended items if we are unlucky.

Reply:
# Summary

TL;DR - we're "safe" afaics.

# Details

I've done a bit of digging and the results are rather interesting...

The `CODEOWNERS` file is rather limited in that it cannot exclude paths meaning that nominally rules like `*.md` would match changes to files like `vendor/github.com/user/project/foo.md`. However, this isn't actually a problem for us. Why? Because we prune non-golang files when vendoring by setting [`non-go=true`](https://github.com/kata-containers/agent/blob/master/Gopkg.toml#L55).

I've double checked all the rules and we are "safe" as long as we continue to filter out non-golang files when we vendor. This is going to be important to check when we switch to using [golang modules](https://github.com/kata-containers/kata-containers/issues/42) rather than using [`dep`](https://github.com/golang/dep).

The kernel rule is also safe in that since it's anchored to the top-level directory of the repo (which could be any of our repos for this version of the `CODEOWNERS` file). That means it cannot match below `vendor/blah/blah/kernel/foo`, so that too is safe.

The `*.proto` rule *looked* to me like we needed to add anchoring to ensure it only matches particular paths too (so it would only match those paths in the agent repo). However, again, since we purge all non-golang files below `vendor/`, we won't have a problem with vendored proto files. By also, we don't need to anchor this rule since we don want to check all `*.proto` file in our repos. An in fact, by *not* anchoring, we suddenly get an extra ack requirement for two repos that don't (but should) have `CODEOWNERS` files or rules for `*.proto` files as they do contain pb files:

- `ksm-throttler`

  https://github.com/kata-containers/ksm-throttler/blob/master/pkg/grpc/ksm.proto

- `runtime`

  https://github.com/kata-containers/runtime/blob/master/protocols/cache/cache.proto